editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+          }
+        });
+      } else {
+        // if there are no local edits, switch to the default version
+        switchVersion(defaultVersion);
+      }
+    } else if (serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+      // if the current version is the default version, switch to the user created version
+      switchVersion(userCreatedVersion);
+    }
+  }
+
+  /**
+   * Switch the active branch version
+   *
+   * @param versionName name of the version to switch to
+   */
+  private void switchVersion(String versionName) {
+    ListenableFuture<Void> switchVersionResult = serviceGeodatabase.switchVersionAsync(versionName);
+    switchVersionResult.addDoneListener(() -> {
+      // check if the active version has switched successfully and update the UI
+      if (serviceGeodatabase.getVersionName().equals(versionName)) {
+        currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+        editFeatureVBox.setDisable(true);
+      } else {
+        new Alert(Alert.AlertType.ERROR, "Error switching version.").show();
+      }
+    });
+  }
+
+  /**
+   * Select a feature if one exists where the user clicked
+   *
+   * @param point location where the user clicked
+   */
+  private void selectFeature(Point2D point){
+    ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, point, 1, false);
+    results.addDoneListener(() -> {
+      try {
+        IdentifyLayerResult layer = results.get();
+        List<GeoElement> identified = layer.getElements();
+        if (!identified.isEmpty()) {
+          GeoElement element = identified.get(0);
+          if (element instanceof ArcGISFeature) {
+            // get the selected feature
+            selectedFeature = (ArcGISFeature) element;
+            featureLayer.selectFeature(selectedFeature);
+            selectedFeature.loadAsync();
+            selectedFeature.addDoneLoadingListener(() -> {
+              if (selectedFeature.getLoadStatus() == LoadStatus.LOADED) {
+                // when a feature has been selected set the damage combo box to the feature's attribute value
+                damageTypeComboBox.getSelectionModel().select((String) selectedFeature.getAttributes().get("typdamage"));
+
+                // enable feature editing UI if not on the default branch version
+                if (!serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+                  editFeatureVBox.setDisable(false);
+                }
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature failed to load.").show();
+              }
+            });
+          }
+        }
+      } catch (InterruptedException | ExecutionException e) {
+        new Alert(Alert.AlertType.ERROR, "Exception getting identify result").show();
```

Comment:
`exception` is a bit of a technical term in a message that's shown to the end user. Perhaps this would be simpler?

```suggestion
        new Alert(Alert.AlertType.ERROR, "Failed to identify the feature").show();
```